browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchFragment.java

```diff
@@ -56,8 +58,52 @@ public void onCreate(Bundle savedInstanceState) {
         } else {
             mRequestCode = Integer.MIN_VALUE;
         }
+        requireActivity().getApplication().registerActivityLifecycleCallbacks(lifecycleCallbacks);
+
     }
 
+    @Override
+    public void onDestroy() {
+        requireActivity().getApplication().unregisterActivityLifecycleCallbacks(lifecycleCallbacks);
+        super.onDestroy();
+    }
+
+    private Application.ActivityLifecycleCallbacks lifecycleCallbacks = new Application.ActivityLifecycleCallbacks() {
+        @Override
+        public void onActivityCreated(Activity activity, Bundle savedInstanceState) { }
+
+        @Override
+        public void onActivityStarted(Activity activity) { }
+
+        @Override
+        public void onActivityResumed(Activity activity) { }
+
+        @Override
+        public void onActivityPaused(Activity activity) { }
+
+        @Override
+        public void onActivityStopped(Activity activity) { }
+
+        @Override
+        public void onActivitySaveInstanceState(Activity activity, Bundle outState) { }
+
+        @Override
+        public void onActivityDestroyed(Activity activity) {
+            // https://stackoverflow.com/questions/39726547/i-want-to-close-chrome-custom-tab-when-action-button-is-clicked
```

Comment:
Note: perhaps it could make sense to have this logic (to bring the activity holding `BraintreeFragment` to the top of the stack) inside the application, instead of the browser-switch sdk. I couldn't think of a scenario where it wouldn't work the way it's implemented in this PR, but I only tested it in our app...